### PR TITLE
Update 0x51-V51-OAuth2 chapter

### PR DESCRIPTION
**Pull request by csfreak92**
Final edits for OAuth 2.0 protocol chapter

This Pull Request relates to updates for issue #996  and a continuation of edits for [pull 1804](https://github.com/OWASP/ASVS/pull/1804). 

Ping @tghosth, @jmanico and @elarlang for further review. I have taken Josh's comments and reviews prior to this PR/commit.

### CODE REVIEW

**Comment by csfreak92** [comment]
The linter is complaining about hard tabs, but what better way to follow that format from this sample https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Terminology ? Any suggestions please?

**Comment by jmanico** [comment]
> The linter is complaining about hard tabs, but what better way to follow that format from this sample https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Terminology ? Any suggestions please?

 To fix this, replace any hard tab characters with spaces instead.

**Comment by csfreak92** [comment]
Fixed the tab issue, but not sure why the linter fails. Looks like some checks are failing.

**Review by jmanico — APPROVED** [review]
Good eye Josh, nice edits

File: 5.0/en/0x51-V51-OAuth2.md [hunk 1]
```diff
@@ -1,17 +1,70 @@
...
+| **51.3.2** | [ADDED] Verify that access tokens are restricted to certain Resource Servers (audience restriction), preferably to a single Resource Server. Every Resource Server is obliged to verify, for every request, whether the access token sent with that request was meant to be used for that particular Resource Server. If not, the Resource Server must refuse to serve the respective request. | ✓ | ✓ | ✓ |
+| **51.3.3** | [ADDED] Verify that access tokens are restricted to certain resources and actions on Resource Servers or resources. Every Resource Server is obliged to verify, for every request, whether the access token sent with that request was meant to be used for that particular action on the particular resource. If not, the Resource Server must refuse to serve the respective request. | ✓ | ✓ | ✓ |
+
+## V51.4 Resource Owner
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **51.4.1** | [ADDED] Verify that the Resource Owner password credentials grant is not used. This grant type insecurely exposes the credentials of the Resource Owner to the client, increasing the attack surface of the application.  | ✓ | ✓ | ✓ |
```
Comment (tghosth) [inline 5.0/en/0x51-V51-OAuth2.md:39]: As per Elar's comment here: https://github.com/OWASP/ASVS/pull/1804#issuecomment-185876662

> 51.4.1 - We can not have requirements for the Resource Owner (end-user). Is a certain flow or grant is accepted/supported, it is the Authorization Server's responsibility to configure it for the OAuth Client. So it belongs to the "Authorization Server" section.

I agree, please address this @csfreak92

**Comment by jmanico** [comment]
For...

+| **51.3.1** | [ADDED] Verify that Resource Servers are using mechanisms for sender-constraining access tokens to prevent token replay, such as Mutual TLS for OAuth 2.0 or OAuth Demonstration of Proof of Possession (DPoP).  | ✓ | ✓ | ✓ |

... you are supposed to use  the same client secret given to the client during registration. This implied it's arbitrary. My edit suggestion is:

+| **51.3.1** | [ADDED] "Verify Resource Servers implement sender-constrained access token mechanisms to mitigate token replay risks. This is achieved by mandating Mutual TLS for OAuth 2.0 or OAuth DPoP, using the client secret provided at client registration.".  | ✓ | ✓ | ✓ |

**Comment by jmanico** [comment]
I merged prematurely. Would you like me to reverse it out? My apologies @tghosth and others.

**Comment by tghosth** [comment]
> I merged prematurely. Would you like me to reverse it out? My apologies @tghosth and others.

I'll ask ralph to handle in the merged version, we are nearly there

**Comment by csfreak92** [comment]
> ... you are supposed to use the same client secret given to the client during registration. This implied it's arbitrary. My edit suggestion is:
> 
> +| **51.3.1** | [ADDED] "Verify Resource Servers implement sender-constrained access token mechanisms to mitigate token replay risks. This is achieved by mandating Mutual TLS for OAuth 2.0 or OAuth DPoP, using the client secret provided at client registration.". | ✓ | ✓ | ✓ |

Noted, @jmanico. I thought of explicitly calling it out as some people might not think it is implied already, but for brevity I agree with this assessment :)

**Comment by tghosth** [comment]
Yes please @csfreak92

**Pull request merged**